### PR TITLE
Added the ability to limit assignments to specific course groups

### LINKED ISSUE #759: Assignments only for specific groups (cohorts)
**Describe the solution you'd like**
As a teacher, I would like to create assignments that are only available for a specific group (Canvas section). This makes it possible to add assignments on cohort basis, whereafter they will only see their track with their own deadlines.

FPS wants to use this

**User story**
Add students from 1 cohort to a Canvas section.
Make an assignment for only that eJournal group (the cohort).
Copy templates and everything from previous year.
Done :)
Now all students from the new cohort can only see that assignment, while still being in the same Canvas bachelor page with all the other students from all the other courses.

### PR DESCRIPTION
**Pull request by Larspolo**
## Description
added cohort based assignments

image
image

## Summary of changes
- Fixes #759 
- security issues

### CODE REVIEW

File: src/django/VLE/models.py [hunk 1]
```diff
@@ -761,6 +783,9 @@ class Journal(models.Model):
         default=0,
     )
 
+    def get_queryset(self):
+        return super(Journal, self).get_queryset().filter(pk=1)
```
Comment (mjvkeulen) [inline src/django/VLE/models.py]: Why is the filter required?

File: src/django/VLE/serializers.py [hunk 2]
```diff
@@ -272,7 +283,7 @@ def get_journals(self, assignment):
         if 'journals' in self.context and 'course' in self.context \
            and self.context['journals'] and self.context['course']:
             course = self.context['course']
-            users = course.participation_set.filter(role__can_have_journal=True).values('user')
+            users = course.participation_set.all().values('user')
```
Comment (mjvkeulen) [inline src/django/VLE/serializers.py]: `All()` can be stripped

File: src/vue/package.json [hunk 3]
```diff
@@ -11,44 +11,44 @@
         "lint": "vue-cli-service lint src --ext .js,.vue"
     },
     "dependencies": {
```
Comment (mjvkeulen) [inline src/vue/package.json]: I would prefer to split dependency updates from isolated feature PR's

File: src/vue/src/views/Guest.vue [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <template>
     <content-single-column>
-        <h1><span>Welcome to {{ name ? name : 'eJournal' }}!</span></h1><br/>
+        <h1><span>Welcome to {{ name ? name : 'eJournal' }}</span></h1><br/>
```
Comment (mjvkeulen) [inline src/vue/src/views/Guest.vue]: Haha good fix

File: src/vue/build/current-release.js [hunk 5]
```diff
@@ -1,6 +1,5 @@
 const execFileSync = require('child_process').execFileSync
-
-const version = execFileSync('git', ['describe', '--abbrev=0', '--tags']).toString().trim()
+const version = execFileSync('git', ['for-each-ref', '--format="%(refname:short)"', '--sort=-authordate', '--count=1', 'refs/tags']).toString().trim()
```
Comment (engelhamer) [inline src/vue/build/current-release.js]: Can you elaborate on this a little?

Comment (Larspolo) [reply src/vue/build/current-release.js]: sort all git tags by date, take the first one

File: src/django/VLE/views/journal.py [hunk 6]
```diff
@@ -50,13 +50,14 @@ def list(self, request):
...
         journals = JournalSerializer(
-            assignment.journal_set,
+            Journal.objects.filter(assignment=assignment),
             many=True,
             context={
                 'user': request.user,
                 'course': course,
             }).data
 
+        print(request.user)
```
Comment (mjvkeulen) [inline src/django/VLE/views/journal.py]: Left over

**Pull request merged**